shelley/chain-and-ledger/executable-spec/test/Examples.hs

```diff
@@ -220,25 +235,41 @@ alicePoolParams =
 
 -- | Helper Functions
 
-mkSeqNonce :: Natural -> Seed
-mkSeqNonce m = foldl (\c x -> c ⭒ Nonce x) NeutralSeed [0..toInteger m]
-
-mkBlock :: HashHeader -> AllPoolKeys -> [Tx] -> Slot
-  -> Seed -> Seed -> UnitInterval -> Natural -> Block
-mkBlock prev pkeys txns s enonce bnonce l kesPeriod =
+mkSeqNonce :: Natural -> Nonce
+mkSeqNonce m = foldl (\c x -> c ⭒ mkNonce x) NeutralNonce [0.. m]
+
+-- | We provide our own nonces to 'mkBlock', which we then wish to recover as
+-- the output of the VRF functions. In general, however, we just derive them
+-- from a natural. Since the nonce is a hash, we do not want to recover it to
+-- find a preimage. In testing, therefore, we just wrap the raw natural, which
+-- we then encode into the fake VRF implementation.
+newtype NatNonce = NatNonce Natural
+
+unitIntervalToNatural :: UnitInterval -> Natural
+unitIntervalToNatural = floor . ((1000 % 1) *) . intervalValue
+
+mkBlock
+  :: HashHeader
+  -> AllPoolKeys
+  -> [Tx]
+  -> Slot
+  -> Nonce
+  -> NatNonce
+  -> UnitInterval
+  -> Natural
+  -> Block
+mkBlock prev pkeys txns s enonce (NatNonce bnonce) l kesPeriod =
   let
     (shot, vhot) = hot pkeys
-    nonceSeed = (enonce ⭒ slotToSeed s) ⭒ SeedEta
-    leaderSeed = (enonce ⭒ slotToSeed s) ⭒ SeedL
+    nonceNonce = (enonce ⭒ slotToNonce s) ⭒ seedEta
```

Comment:
this is related to my comment above, that we need a second/XOR operation for the two seed constants.